gc/api/include/OMR/GC/Handle.hpp

```diff
@@ -0,0 +1,151 @@
+/*******************************************************************************
+ *  Copyright (c) 2018, 2018 IBM and others
+ *
+ *  This program and the accompanying materials are made available under
+ *  the terms of the Eclipse Public License 2.0 which accompanies this
+ *  distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ *  or the Apache License, Version 2.0 which accompanies this distribution and
+ *  is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ *  This Source Code may also be made available under the following
+ *  Secondary Licenses when the conditions for such availability set
+ *  forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ *  General Public License, version 2 with the GNU Classpath
+ *  Exception [1] and GNU General Public License, version 2 with the
+ *  OpenJDK Assembly Exception [2].
+ *
+ *  [1] https://www.gnu.org/software/classpath/license.html
+ *  [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ *  SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#if !defined(OMR_GC_HANDLE_HPP_)
+#define OMR_GC_HANDLE_HPP_
+
+#include <OMR/TypeTraits.hpp>
+#include <cstddef>
+#include <omrcfg.h>
+
+namespace OMR
+{
+namespace GC
+{
+
+template <typename T>
+class HandleBase
+{
+public:
+	HandleBase() {}
+
+	HandleBase(T* const volatile* root) : _root(root) {}
+
+	/// Convert this Handle to a raw indirect pointer. The address of the rooting slot.
```

Comment:
Why are all of the comments using `///` instead of `//` or `/* .. */`?

Reply:
They're meant to be doxygen comments.

Reply:
Although, after skimming the doxygen documentation, it looks like these comments need to be at least 2 lines before doxygen will pick them up? Should I change to javadoc-style comments? eg
```c++
/**
 * Convert this Handle to a raw indirect pointer. The address of the rooting slot.
 */
```

Reply:
Yes I would convert them to the javadoc style.  Thanks